pkg/daemon/daemon.go

```diff
@@ -1351,7 +1351,10 @@ func getRefDigest(ref string) (string, error) {
 	return canon.Digest().String(), nil
 }
 
-// compareOSImageURL is the backend for checkOS.
+// compareOSImageURL is the backend for checkOS.  This function
+// used to have more sophistication around comparing the digests,
+// but we removed that.  The dummy:// could also likely go away.
+// https://github.com/openshift/machine-config-operator/issues/1964
 func compareOSImageURL(current, desired string) (bool, error) {
```

Comment:
We can remove the error return from the function altogether at this point.
```golang
func compareOSImageURL(current, desired string) bool {
	return desired == "" || desired == "://dummy" || current == desired
```
("No target" log omitted for demonstration only)

Reply:
Yep you're right, it turns into a bigger patch but as long as we're here we might as well do the bigger cleanup.